include/solvers/simulation_parameters.h

```diff
@@ -59,7 +59,7 @@ class SimulationParameters
   AnalyticalSolutions::AnalyticalSolution<dim> *    analytical_solution;
   SourceTerms::SourceTerm<dim> *                    source_term;
   Parameters::VelocitySource                        velocity_sources;
-  Parameters::IBParticles<dim>                      particlesParameters;
+  Parameters::IBParticles<dim> *                    particlesParameters;
```

Comment:
Good idea here would be to transition this Parameter and the other parameters to shared pointers. It would be a small change and would make things cleaner in the code.